docs/sources/tempo/setup/upgrade.md

```diff
@@ -13,7 +13,125 @@ The upgrade process changes for each version, depending upon the changes made fo
 
 This upgrade guide applies to on-premise installations and not for Grafana Cloud.
 
->**TIP**: You can check your configuration options using the [`status` API endpoint]({{< relref "../api_docs#status" >}}) in your Tempo installation.
+{{% admonition type="tip" %}}
+You can check your configuration options using the [`status` API endpoint]({{< relref "../api_docs#status" >}}) in your Tempo installation.
+{{% /admonition %}}
+
+## Upgrade to Tempo 2.4
+
+Tempo 2.4 has several considerations for any upgrade:
+
+* vParquet3 is now the default backend
+* Caches configuration was refactored
+* Updated, removed, and renamed configuration parameters
+
+For a complete list of changes, enhancements, and bug fixes, refer to the [Tempo 2.4 changelog](https://github.com/grafana/tempo/releases).
+
+### Transition to vParquet3 as default block format
+
+vParquet3 format is now the default block format. It is production ready and we highly recommend switching to it for improved query performance and [dedicated attribute columns]({{< relref "../operations/dedicated_columns" >}}).
+
+Upgrading to Tempo 2.4 modifies the Parquet block format. Although you can use Tempo 2.3 with vParquet2 or vParquet3, you can only use Tempo 2.4 with vParquet3.
+
+With this release, vParquet is being deprecated. Tempo 2.4 will still read vParquet1 blocks. However, Tempo will fail to configure them. [[PR 3377](https://github.com/grafana/tempo/pull/3377/files#top)]
+
+For information on changing the vParquet version, refer to [Choose a different block format](https://grafana.com/docs/tempo/next/configuration/parquet#choose-a-different-block-format).
+
+### Cache configuration refactored
+
+The major cache refactor to allow multiple role-based caches to be configured. [[PR 3166](https://github.com/grafana/tempo/pull/3166)]
+This change resulted in the following fields being deprecated.
+These have all been migrated to a top level `cache:` field.
+
+For more information about the configuration, refer to the [Cache]({{< relref "../configuration#cache" >}}) section.
+
+The old configuration block looked like this:
+
+```yaml
+storage:
+  trace:
+    cache:
+    search:
+      cache_control:
+    background_cache:
+    memcached:
+    redis:
+```
+
+With the new configuration, you create your list of caches,- with either `redis` or `memcached` cluster with your config, then define the types of data and roles.
+
+```yaml
+cache:
```

Comment:
same thought as above

Reply:
Because this is the upgrade page, what do you think about including a simple example and then the longer example? Or mabe the longer example should be in the configuration documentation?

Reply:
I put the short example in the upgrade page as a "simple" example and kept the longer example as a "configuration example showing more options."

Reply:
I think we should just remove it. An operator can look up the various config options for memcached/redis independently. Adding them just feels like it confuses the message which is to just show the structure of the new cache block.

also some of those settings are borderline misleading. e.g. "service" actually tells tempo to look for a named port which i always found weird.

Reply:
okay. I'll remove it.